### PR TITLE
Switch PostgreSQL container to use pgautoupgrade

### PR DESCRIPTION
**Pull request by Daniel15**
Lemmy changed from PostgreSQL 15 to 16 at some point, and trying to switch from 15 to 16 in `docker-compose.yml` results in errors like this at startup:

> DETAIL:  The data directory was initialized by PostgreSQL version 15, which is not compatible with this version 16.3

This PR changes the PostgreSQL container to use [pgautoupgrade](https://github.com/pgautoupgrade/docker-pgautoupgrade). The pgautoupgrade image is based on the regular PostgreSQL one, but it automatically updates the data to the newer format on startup. This avoids users having to [figure it out manually](https://github.com/docker-library/postgres/issues/37).

### CODE REVIEW

**Comment by dessalines** [comment]
Thx! I had no idea this existed, and I love it.

**Review by dessalines — APPROVED** [review]

**Comment by Nutomic** [comment]
Looks good, please change the one in `docker/federation/docker-compose.yml` as well.

**Comment by ticoombs** [comment]
`scripts/postgres_12_to_15_upgrade.sh` would be completely redundant after this change fyi.

**Comment by Daniel15** [comment]
I'll make the requested changes soon.



> `scripts/postgres_12_to_15_upgrade.sh` would be completely redundant after this change fyi.

The current docker-compose.yml uses PostgreSQL 16 so this script seems out-of-date anyways.

**Comment by justinclift** [comment]
As a data point, we've added Debian based pgautoupgrade images now too (`15-bookworm` and `16-bookworm`).

If your existing PostgreSQL docker image is Debian based, then it's best to use a Debian based pgautoupgrade image.

That's because we've had reports of collation problems when people upgraded their Debian based PostgreSQL image using our original (Alpine based) images.  These new images should fix that. :smile:

**Comment by justinclift** [comment]
Heh, and now I've looked at the actual PR (a simple 1-liner), and it's alpine based anyway.  You should be good to go. :smile:

**Comment by dessalines** [comment]
I've fixed the other cases now.

**Comment by Daniel15** [comment]
> I've fixed the other cases now.

Thanks. Sorry I didn't get to it - I've been busy with other things.

**Comment by dessalines** [comment]
No probs.

**Review by ticoombs — APPROVED** [review]
Approving under the condition we have tested in CI and its passing.

File: scripts/postgres_15_to_16_upgrade.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ echo "Removing the old postgres folder"
 sudo rm -rf volumes/postgres
 
 echo "Updating docker compose to use postgres version 16."
-sudo sed -i "s/image: .*postgres:.*/image: docker.io\/postgres:16-alpine/" ./docker-compose.yml
+sudo sed -i "s/image: .*postgres:.*/image: pgautoupgrade\/pgautoupgrade:16-alpine/" ./docker-compose.yml
 
 echo "Starting up new postgres..."
 sudo docker compose up -d postgres
```
Comment (Nutomic) [inline scripts/postgres_15_to_16_upgrade.sh:30]: As @ticoombs mentioned, these migration scripts arent needed anymore. However they are still linked in Lemmy release announcements, so I would leave them in but remove everything except for this line:
`sudo sed -i "s/image: .*postgres:.*/image: pgautoupgrade\/pgautoupgrade:16-alpine/" ./docker-compose.yml`

Comment (dessalines) [reply scripts/postgres_15_to_16_upgrade.sh]: Done

**Review by Nutomic — APPROVED** [review]

**Pull request merged**